### PR TITLE
Increase test coverage for server InstanceImpl

### PR DESCRIPTION
**Pull request by dnoe**
The failed-to-find case of InstanceImpl::getListenSocketFd() was
previously not covered by testing.  This change introduces fixture &
mocking changes required to unit test the server itself (InstanceImpl)
and create a test case that exercises the return -1 case in
getListenSocketFd().

### CODE REVIEW

File: test/mocks/server/mocks.cc [hunk 1]
```diff
@@ -8,7 +8,11 @@ using testing::SaveArg;
 
 namespace Server {
 
-MockOptions::MockOptions() {}
+MockOptions::MockOptions() : path_("test/config/integration/server.json") {
```
Comment (htuch) [inline test/mocks/server/mocks.cc]: You should probably set this inside the test, since different tests may want different path_ values.

**Comment by mattklein123** [comment]
@dnoe please sign CLA if you didn't already do it.

File: test/mocks/server/mocks.cc (hunk 1 above)
Comment (dnoe) [inline test/mocks/server/mocks.cc]: Good idea, I will update.

**Comment by dnoe** [comment]
CLA is signed, and I believe I have updated this pull request to parameterize the config path as requested.  I am more familiar with the gerrit workflow so please let me know if I have done anything weird with the PR.

**Comment by htuch** [comment]
LGTM. A tip for future reviews: I think you might have done a force push to update - it's better to not rewrite history on the GitHub branch and instead provide additional increment commits and push the branch cleanly. This lets the GitHub review system keep track of which comments are stale and allows reviewers to see the diff between commits. It's OK to do a force push before folks have started to review though. The eventual merge rolls up all commits into a single commit.

File: test/mocks/server/mocks.cc [hunk 2]
```diff
@@ -8,7 +8,11 @@ using testing::SaveArg;
 
 namespace Server {
 
-MockOptions::MockOptions() {}
+MockOptions::MockOptions(const std::string& path) : path_(path) {
+  ON_CALL(*this, fileFlushIntervalMsec()).WillByDefault(Return(std::chrono::milliseconds(1000)));
+  ON_CALL(*this, restartEpoch()).WillByDefault(Return(0));
```
Comment (mattklein123) [inline test/mocks/server/mocks.cc]: nit: IIRC this is default, so can be removed.

File: test/mocks/server/mocks.h [hunk 3]
```diff
@@ -27,7 +27,9 @@ namespace Server {
 
 class MockOptions : public Options {
 public:
-  MockOptions();
+  // If it is not provided the configPath will be an empty string in the mock.
```
Comment (mattklein123) [inline test/mocks/server/mocks.h]: nit: s/it/path (though I think this is kind of obvious and comment can probably be removed)

File: test/server/server_test.cc [hunk 4]
```diff
@@ -2,6 +2,10 @@
 
 #include "test/mocks/common.h"
 #include "test/mocks/init/mocks.h"
+#include "test/mocks/server/mocks.h"
+#include "test/mocks/stats/mocks.h"
+
+#include "test/integration/server.h" // for TestDrainManager, TestIsolatedStoreImpl
```
Comment (mattklein123) [inline test/server/server_test.cc]: nit: comment not needed, can we alpha order and move above line 3

File: test/server/server_test.cc [hunk 5]
```diff
@@ -29,4 +33,42 @@ TEST(InitManagerImplTest, Targets) {
...
+  testing::NiceMock<MockHotRestart> restart_;
+  Stats::TestIsolatedStoreImpl stats_store_;
+  Thread::MutexBasicLockable fakelock_;
+  TestComponentFactory component_factory_;
+  testing::NiceMock<LocalInfo::MockLocalInfo> local_info_;
+  InstanceImpl server_;
+};
+
+TEST_F(ServerInstanceImplTest, NoListenSocketFds) {
+  std::string bad_tcp_url("tcp://255.255.255.255:80");
```
Comment (mattklein123) [inline test/server/server_test.cc]: nit: local var not needed, just pass to function directly

**Comment by dnoe** [comment]
Nit picks addressed in a second commit that is now part of this PR.

**Pull request merged**